cloudify_hostpool/storage/sqlite.py

```diff
@@ -57,35 +59,41 @@ def add_host(self, host):
             cursor.execute('INSERT INTO {0} (host_id, host, '
                            'public_address, auth, port, alive, reserved)'
                            ' VALUES(?, ?, ?, ?, ?, ?, ?)'
-                           .format(SQLiteStorage._TABLE_NAME), values)
+                           .format(_SQLiteStorageBase._TABLE_NAME), values)
             host['global_id'] = cursor.lastrowid
 
     def update_host(self, global_id, host):
         if not host:
             return None
-        try:
-            with sqlite3.connect(self._filename, isolation_level='EXCLUSIVE')\
-                    as conn:
-                conn.row_factory = _dict_factory
-                conn.execute('BEGIN EXCLUSIVE')
-                cursor = conn.cursor()
-                values = tuple(host.itervalues())
-                sql_part = ", ".join('{0}=?'.format(s) for s in host)
-                cursor.execute('SELECT * FROM {0} WHERE global_id=?'
-                               .format(SQLiteStorage._TABLE_NAME),
-                               (global_id, ))
-                hst = cursor.fetchone()
-                if all(item in hst.iteritems() for item in host.iteritems()):
-                    return None
-                cursor.execute('UPDATE {0} SET {1} WHERE global_id=?'
-                               .format(SQLiteStorage._TABLE_NAME, sql_part),
-                               values + (global_id,))
-                hst.update(host)
-                return hst
-        except sqlite3.OperationalError as e:
-            if e.message == 'database is locked':
-                raise DBLockedError()
-            raise DBError(e.message)
+        while True:
+            try:
+                with sqlite3.connect(self._filename,
+                                     isolation_level='EXCLUSIVE') as conn:
+                    conn.row_factory = _dict_factory
+                    conn.execute('BEGIN EXCLUSIVE')
+                    cursor = conn.cursor()
+                    values = tuple(host.itervalues())
+                    sql_part = ", ".join('{0}=?'.format(s) for s in host)
+                    cursor.execute('SELECT * FROM {0} WHERE global_id=?'
+                                   .format(_SQLiteStorageBase._TABLE_NAME),
+                                   (global_id, ))
+                    hst = cursor.fetchone()
+                    if all(item in hst.iteritems()
+                           for item in host.iteritems()):
+                        return None
+                    cursor.execute('UPDATE {0} SET {1} WHERE global_id=?'
+                                   .format(_SQLiteStorageBase._TABLE_NAME,
+                                           sql_part),
+                                   values + (global_id,))
+                    hst.update(host)
+                    return hst
+            except sqlite3.OperationalError as e:
+                if e.message == 'database is locked':
+                    if self.blocking:
```

Comment:
This was intentional. Reservation returns if a lock had been obtained or not - there is no third option - it is blocking by definition.